srtcore/api.cpp

```diff
@@ -2387,18 +2381,14 @@ void CUDTUnited::removeSocket(const SRTSOCKET u)
 
 void CUDTUnited::setError(CUDTException* e)
```

Comment:
The `setError` function should be now marked static. It would be tempting to replace all calls to `s_UDTUnited.setError`, but there are too many of them and it would make an unnecessary noise. Still, this function no longer needs access to fields.

In future we can think about a refactoring that will make the `CUDTException` object permanent, instead of being allocated and deallocated. This requires however a change at the place where `setError` is called (so that `new` is no longer used), so this would be a good opportunity to kill two birds with one stone.

Reply:
Removed `CUDTUnited::setError(..)`